src/sentry/models/file.py

```diff
@@ -355,7 +355,18 @@ def save_to(self, path):
             f = self._get_chunked_blob(
                 prefetch=True, prefetch_to=base, delete=False
             ).detach_tempfile()
-            os.rename(f.name, path)
+
+            # pre-emptively check if the file already exists.
+            # this can happen as a race condition if two processes/threads
+            # are trying to cache the same file and both try to write
+            # at the same time, overwriting each other. Normally this is fine,
+            # but can cause an issue if another process has opened the file
+            # for reading, then the file that was being read gets clobbered.
+            # I don't know if this affects normal filesystems, but it
+            # definitely has an issue if the filesystem is NFS.
+            if not os.path.exists(path):
```

Comment:
Should we have an `else` removing the file so it doesn't just stick around in tmp?

Reply:
It's already being done in `finally` path.

Reply:
Oh, I see. I guess it would be orphaned. Lemme fix up.

Reply:
Fixed so it does rely on the finally path now.